### PR TITLE
Always use the openssl backend of curl

### PR DESCRIPTION
**Pull request by saiarcot895**
This should make sure that the openssl backend of curl is used, instead of the gnutls or nss backend, for consistency reasons.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211

### CODE REVIEW

**Review by ganglyu — APPROVED** [review]

**Review by xumia — APPROVED** [review]

File: sonic-slave-bullseye/Dockerfile.j2 [hunk 1]
```diff
@@ -202,7 +202,7 @@ RUN apt-get update && apt-get install -y \
         libxml2-dev \
 # For BFN sdk build
         libusb-1.0-0-dev \
-        libcurl3-nss-dev \
+        libcurl4-openssl-dev \
```
Comment (xumia) [inline sonic-slave-bullseye/Dockerfile.j2:205]: Can we change the one in sonic-slave-buster/Dockerfile.j2 as well? We are still using the docker image.

Comment (saiarcot895) [reply sonic-slave-bullseye/Dockerfile.j2:205]: Done

**Comment by saiarcot895** [comment]
@k-v1 I think it would be better for feature/capability consistency (and some minor space savings) that everything uses the openssl version of curl, unless there's a very good reason that some other backend needs to be used.

**Comment by bluecmd** [comment]
I filed https://github.com/sonic-net/sonic-mgmt-framework/issues/130 to investigate changing it to openssl as well instead of gnutls.

**Pull request merged**